backends-velox/src/main/scala/org/apache/spark/sql/hive/execution/HiveFileFormat.scala

```diff
@@ -89,13 +92,11 @@ class HiveFileFormat(fileSinkConf: FileSinkDesc)
         .get("spark.plugins")
         .equals("io.glutenproject.GlutenPlugin")
     ) {
-      val compressionCodec = if (fileSinkConf.compressed) {
-        fileSinkConf.compressCodec
-      } else {
-        "none"
-      }
+      val conf = ContextUtil.getConfiguration(job)
+      val parquetOptions = new ParquetOptions(options, sparkSession.sessionState.conf)
+      conf.set(ParquetOutputFormat.COMPRESSION, parquetOptions.compressionCodecClassName)
```

Comment:
Have you tried add config `spark.hadoop.hive.exec.compress.output=true` ? Since it's hive write, it should follow the hive behavior. We can put the additional compression codec to `else` branch to replace `none`.

Reply:
Yes. Put the spark sql related compression codec into the else branch. Please help to review again. Thanks.